### PR TITLE
feat: add python 3.11 support

### PR DESCRIPTION
**Pull request by ichintanjoshi**
## Description 

This PR contains changes for upgrading python 3.8 to python 3.11.

Changes include :-
- Dependencies and version upgrades
- Extra check for fractional function to get only integers
    - This is done because `fractions` function in miller.py file used `fr.gcd`
    - Since python 3.9: The [math.gcd()](https://docs.python.org/3/library/math.html#math.gcd) function is now used to normalize the numerator and denominator. [math.gcd()](https://docs.python.org/3/library/math.html#math.gcd) always return a [int](https://docs.python.org/3/library/functions.html#int) type. Previously, the GCD type depended on numerator and denominator

Done as a part of following :-
- https://github.com/openedx/public-engineering/issues/233
- https://github.com/openedx/edx-platform/issues/34229
- https://github.com/openedx/openedx-chem/issues/77

### CODE REVIEW

**Comment by ichintanjoshi** [comment]
@feanil  The tests are green now.

**Comment by ichintanjoshi** [comment]
@UsamaSadiq  have updated accordingly

File: chem/__init__.py [hunk 1]
```diff
@@ -1,3 +1,3 @@
 """ init """
 
-__version__ = '1.2.0'
+__version__ = '1.2.1'
```
Comment (feanil) [inline chem/__init__.py:3]:
```suggestion
__version__ = '1.3.0'
```

We should do a minor version bump for this change.

**Review by feanil — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@ichintanjoshi 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-chem&entry.867055334=https://github.com/openedx/openedx-chem/pull/78&entry.1484655318=https://github.com/ichintanjoshi&entry.752974735=2024-03-28+12:53&entry.1917517419=2024-04-22+14:25&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**